go/vt/mysqlctl/azblobbackupstorage/azblob.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2020 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package azblobbackupstorage implements the BackupStorage interface
+// for Azure Blob Storage
+package azblobbackupstorage
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"vitess.io/vitess/go/vt/concurrency"
+	"vitess.io/vitess/go/vt/log"
+	"vitess.io/vitess/go/vt/mysqlctl/backupstorage"
+)
+
+var (
+	// This is the account name
+	accountName = flag.String("azblob_backup_account_name", "", "Azure Account Name for Backups")
+
+	// This is the private access key
+	accountKey = flag.String("azblob_backup_account_key", "", "Azure Account Key")
```

Comment:
I don't like that the key itself will be baked into the command line. Does Azure have a standard file format for credentials (ideally both account name and key)? If not, can we make this flag a path to a file containing the key?

Reply:
There is no standard file format for the credentials. There are some things stored at `$HOME/.azure`, but the credentials for the storage containers are not one of them. Those are special purpose keys that are not part of anyone's storage account

Reply:
I would just ask for a file with just the key in it, then.

Reply:
I have also added environmental paramaters `VITESS_AZBLOB_ACCOUNT_{NAME,KEY} to prevent these from having to be on the command line. This is what minio does as well